modules/sync/syncs/urlsynchronization.cpp

```diff
@@ -253,4 +285,103 @@ std::string UrlSynchronization::generateUid() {
     return _identifier;
 }
 
+bool UrlSynchronization::isEachFileValid() {
+    std::filesystem::path path = directory();
+    path.replace_extension("ossync");
+    // Check if file exists at all
+    if (!std::filesystem::is_regular_file(path)) {
+        return false;
+    }
+
+    // Read contents of file
+    std::ifstream file(path);
+    std::string line;
+
+    file >> line;
+    // Update ossync files that does not have a version number to new format
+    if (line == SynchronizationToken) {
+        if (_secondsUntilResync == 0) {
+            return false; // Force download new file
+        }
+
+        // We must close file early because createSyncFile writes to it
+        file.close();
+        createSyncFile();
+        // File is valid until some date
+        return true; 
+    }
+    // Otherwise first line is the version number.
+    std::string ossyncVersion = line;
+
+    //Format of 1.0 ossync:
+    //Version number: e.g., 1.0
+    //Date that specifies how long the files are valid for in ISO8601 format
+    //Valid to: yyyy-mm-ddThr:mn:sc.xxx
+
+    if (ossyncVersion == "1.0") {
+        std::getline(file >> std::ws, line);
+        std::string& fileIsValidToDate = line;
+        double fileValidAsJ2000 = Time::convertTime(fileIsValidToDate);
+
+        std::string todaysDate = Time::currentWallTime();
+        double todaysDateAsJ2000 = Time::convertTime(todaysDate);
+
+        // Issue warning if file is kept but user changed setting to download on startup.
+        if ((fileValidAsJ2000 > todaysDateAsJ2000) && _secondsUntilResync == 0) {
+            LWARNING(fmt::format(
+                "{}: File is valid to {} but asset specifies SecondsUntilResync = {} "
+                "Did you mean to re-download the file? If so, remove file from sync "
+                "folder to resync",
+                _identifier,
+                fileIsValidToDate,
+                _secondsUntilResync
+            ));
+        }
+
+        // Returns true if date in ossync file is 'larger' than todays date,
+        // i.e. no sync needed.
+        return fileValidAsJ2000 > todaysDateAsJ2000;
+    }
+    else {
+        LERROR(fmt::format(
+            "{}: Unknown ossync version number read. "
```

Comment:
We're not as strict with the "one parameter per line" for log message function calls. Looks like this could be written something like: 

```
        LERROR(fmt::format(
            "{}: Unknown ossync version number read. Got {} while {} and below are valid",
            _identifier, ossyncVersion, OssyncVersionNumber
        ));
```

and still stay within the allowed line length. And it's more compact and easier to read :)

Reply:
I always mixed this up;  disregard my previous comment @engbergandreas  :D